File: Arrays/contains_duplicate/contains_duplicate.py



if __name__ == "__main__":
    Main()

def brute_force(array):

    for i in range(len(array)):
        for j in range(i + 1, len(array)):
            if nums[i] == nums[j]:
                return True

    return False

def hashmap(array):

    hashmap = set()

    for x in array:
        if x in hashmap:
            return True

    return False

def Main():

    test_case_1 = [1, 2, 3, 4, 5, 6]
    test_case_2 = [1, 2, 2, 3, 4, 5]

    print(f'test case 1: {test_case_1}')
    print(f'test case 2: {test_case_2}')
    print(f'brute force: {brute_force(test_case_1)}')
    print(f'brute force: {brute_force(test_case_2)}')
    print(f'hashmap: {hashmap(test_case_1)}')
    print(f'hashmap: {hashmap(test_case_2)}')

    

File: Arrays/contains_duplicate/contains_duplicate.js


function brute_force(array) {
    for (let i = 0; i < array.length; i++) {
        for (let j = i + 1; j < array.length; j++) {
            if (array[i] == array[j]) {
                return true;
            }
        }
    }
    return false;
}

function hash_set(array) {

    let hashset = new Set();

    for (let i = 0; i < array.length; i++) {
        if (hashset.has(array[i])) {
            return true;
        }
        hashset.add(array[i])
    }
    return false;
}

let test_case_1 = [1, 2, 3, 4, 5, 6];
let test_case_2 = [1, 2, 2, 3, 4, 5];

console.log('test case 1 :', test_case_1);
console.log('test case 2 :', test_case_2);

console.log('brute_force 1: ', brute_force(test_case_1));
console.log('brute_force 2: ', brute_force(test_case_2));

console.log('hashset 1: ', hash_set(test_case_1));
console.log('hashset 2: ', hash_set(test_case_2));
